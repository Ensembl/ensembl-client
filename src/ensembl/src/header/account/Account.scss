@import 'src/styles/common';

.account {
  border-bottom: 1px solid $ens-medium-grey;
  height: 300px;
  @include flex;
  @include flex-align(center, middle);

  h2 {
    color: $ens-grey;
    text-align: center;
  }
}
