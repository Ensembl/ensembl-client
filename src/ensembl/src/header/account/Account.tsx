import React, { FunctionComponent, ReactNode } from 'react';
import SlideDown from 'react-slidedown';
import { connect } from 'react-redux';

import { RootState } from 'src/rootReducer';
import { getAccountExpanded } from '../headerSelectors';

import styles from './Account.scss';

type StateProps = {
  accountExpanded: boolean;
};

type OwnProps = {};

type AccountProps = StateProps & OwnProps;

export const Account: FunctionComponent<AccountProps> = (
  props: AccountProps
) => {
  const AccountChildren: ReactNode = (
    <div className={styles.account}>
      <h2>Account area placeholder</h2>
    </div>
  );

  return (
    <SlideDown>{props.accountExpanded ? AccountChildren : null}</SlideDown>
  );
};

const mapStateToProps = (state: RootState): StateProps => ({
  accountExpanded: getAccountExpanded(state)
});

export default connect(mapStateToProps)(Account);
