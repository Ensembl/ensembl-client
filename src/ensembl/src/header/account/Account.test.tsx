import React from 'react';
import { shallow } from 'enzyme';

import { Account } from './Account';

describe('<Account />', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<Account accountExpanded={true} />);
  });

  describe('when account is', () => {
    test('expanded renders children', () => {
      expect(wrapper.find('.account')).toHaveLength(1);
    });

    test('collapsed does not render children', () => {
      wrapper.setProps({ accountExpanded: false });
      expect(wrapper.find('.account')).toHaveLength(0);
    });
  });
});
