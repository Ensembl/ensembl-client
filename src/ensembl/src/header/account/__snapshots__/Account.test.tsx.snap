// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Account /> renders correctly 1`] = `
<ForwardRef>
  <div>
    <h2>
      Account area placeholder
    </h2>
  </div>
</ForwardRef>
`;
